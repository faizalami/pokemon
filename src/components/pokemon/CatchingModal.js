/** @jsxImportSource @emotion/react */
import BaseModal from '../BaseModal';
import { css, keyframes } from '@emotion/react';
import { Flex } from '../FlexGrid';
import Image from '../Image';

const pokeballFly = keyframes`
  from {
    left: 0;
    right: 100%;
    transform: rotate(0deg);
  }
  to {
    right: 0;
    transform: rotate(359deg);
  }
`;

const pokeball = css`
  animation: ${pokeballFly} 1s;
  animation-iteration-count: infinite;
  position: absolute;
  top: 0;
  z-index: 1;
`;

const pikachuShake = keyframes`
  0% {
    transform: rotate(5deg);
  }
  25% {
    transform: rotate(0deg);
  }
  50% {
    transform: rotate(-5deg);
  }
  100% {
    transform: rotate(0deg);
  }
`;

const pikachu = css`
  animation: ${pikachuShake} 1s;
  animation-iteration-count: infinite;
`;

const animationContainer = css`
  width: 80%;
  height: 64px;
  position: relative;
`;

function CatchingModal () {
  return (
    <BaseModal>
      <Flex justifyContent="space-between" alignItems="center" css={animationContainer}>
        <Image
          png="/assets/img/png/pokeball-64.png"
          webp="/assets/img/webp/pokeball-64.webp"
          alt="Pokeball"
          lazy
          width={64}
          height={64}
          css={pokeball}
        />
        <Image
          png="/assets/img/png/pikachu-64.png"
          webp="/assets/img/webp/pikachu-64.webp"
          alt="Pokemon"
          lazy
          width={56}
          height={56}
          css={pikachu}
        />
      </Flex>
      <h2>Catching Pokemon . . .</h2>
    </BaseModal>
  );
}

export default CatchingModal;
