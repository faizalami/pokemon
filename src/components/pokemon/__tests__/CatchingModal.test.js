import { render } from '@testing-library/react';
import CatchingModal from '../CatchingModal';

test('catching modal matches snapshot', () => {
  expect(render(<CatchingModal/>)).toMatchSnapshot();
});
