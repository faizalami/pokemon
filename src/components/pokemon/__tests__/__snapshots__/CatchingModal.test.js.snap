// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`catching modal matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": @keyframes animation-0 {
  from {
    left: 0;
    right: 100%;
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  to {
    right: 0;
    -webkit-transform: rotate(359deg);
    -moz-transform: rotate(359deg);
    -ms-transform: rotate(359deg);
    transform: rotate(359deg);
  }
}

@keyframes animation-1 {
  0% {
    -webkit-transform: rotate(5deg);
    -moz-transform: rotate(5deg);
    -ms-transform: rotate(5deg);
    transform: rotate(5deg);
  }

  25% {
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  50% {
    -webkit-transform: rotate(-5deg);
    -moz-transform: rotate(-5deg);
    -ms-transform: rotate(-5deg);
    transform: rotate(-5deg);
  }

  100% {
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  position: fixed;
  top: 0;
  left: 0;
  height: 100%;
  z-index: 100;
  background-color: rgba(0, 0, 0, 0.5);
  width: 100%;
}

.emotion-1 {
  max-width: 48rem;
  background-color: white;
  border-top: 2rem solid #E53935;
  border-bottom: 2rem solid #303C42;
  width: 100%;
  border-radius: 0.5rem;
  margin-top: auto;
  margin-bottom: auto;
  margin-left: 2rem;
  margin-right: 2rem;
}

@media screen and (min-width: 1024px) {
  .emotion-1 {
    margin-left: auto;
    margin-right: auto;
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: white;
  margin: -1rem 0;
  padding: 1rem;
  width: 100%;
  border-radius: 0.5rem;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 80%;
  height: 64px;
  position: relative;
}

.emotion-4 {
  object-fit: contain;
  object-position: center;
  -webkit-animation: animation-0 1s;
  animation: animation-0 1s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  position: absolute;
  top: 0;
  z-index: 1;
}

.emotion-5 {
  object-fit: contain;
  object-position: center;
  -webkit-animation: animation-1 1s;
  animation: animation-1 1s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
}

<body>
    <div>
      <div
        class="emotion-0"
      >
        <div
          class="emotion-1"
        >
          <div
            class="emotion-2"
          >
            <div
              class="emotion-3"
            >
              <picture>
                <source
                  data-testid="image-webp"
                  srcset="/assets/img/webp/pokeball-64.webp"
                  type="image/webp"
                />
                <source
                  data-testid="image-png"
                  srcset="/assets/img/png/pokeball-64.png"
                  type="image/png"
                />
                <img
                  alt="Pokeball"
                  class="emotion-4"
                  data-testid="image-regular"
                  height="64"
                  loading="lazy"
                  src="/assets/img/png/pokeball-64.png"
                  width="64"
                />
              </picture>
              <picture>
                <source
                  data-testid="image-webp"
                  srcset="/assets/img/webp/pikachu-64.webp"
                  type="image/webp"
                />
                <source
                  data-testid="image-png"
                  srcset="/assets/img/png/pikachu-64.png"
                  type="image/png"
                />
                <img
                  alt="Pokemon"
                  class="emotion-5"
                  data-testid="image-regular"
                  height="56"
                  loading="lazy"
                  src="/assets/img/png/pikachu-64.png"
                  width="56"
                />
              </picture>
            </div>
            <h2>
              Catching Pokemon . . .
            </h2>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": @keyframes animation-0 {
  from {
    left: 0;
    right: 100%;
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  to {
    right: 0;
    -webkit-transform: rotate(359deg);
    -moz-transform: rotate(359deg);
    -ms-transform: rotate(359deg);
    transform: rotate(359deg);
  }
}

@keyframes animation-1 {
  0% {
    -webkit-transform: rotate(5deg);
    -moz-transform: rotate(5deg);
    -ms-transform: rotate(5deg);
    transform: rotate(5deg);
  }

  25% {
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  50% {
    -webkit-transform: rotate(-5deg);
    -moz-transform: rotate(-5deg);
    -ms-transform: rotate(-5deg);
    transform: rotate(-5deg);
  }

  100% {
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  position: fixed;
  top: 0;
  left: 0;
  height: 100%;
  z-index: 100;
  background-color: rgba(0, 0, 0, 0.5);
  width: 100%;
}

.emotion-1 {
  max-width: 48rem;
  background-color: white;
  border-top: 2rem solid #E53935;
  border-bottom: 2rem solid #303C42;
  width: 100%;
  border-radius: 0.5rem;
  margin-top: auto;
  margin-bottom: auto;
  margin-left: 2rem;
  margin-right: 2rem;
}

@media screen and (min-width: 1024px) {
  .emotion-1 {
    margin-left: auto;
    margin-right: auto;
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: white;
  margin: -1rem 0;
  padding: 1rem;
  width: 100%;
  border-radius: 0.5rem;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 80%;
  height: 64px;
  position: relative;
}

.emotion-4 {
  object-fit: contain;
  object-position: center;
  -webkit-animation: animation-0 1s;
  animation: animation-0 1s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  position: absolute;
  top: 0;
  z-index: 1;
}

.emotion-5 {
  object-fit: contain;
  object-position: center;
  -webkit-animation: animation-1 1s;
  animation: animation-1 1s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
}

<div>
    <div
      class="emotion-0"
    >
      <div
        class="emotion-1"
      >
        <div
          class="emotion-2"
        >
          <div
            class="emotion-3"
          >
            <picture>
              <source
                data-testid="image-webp"
                srcset="/assets/img/webp/pokeball-64.webp"
                type="image/webp"
              />
              <source
                data-testid="image-png"
                srcset="/assets/img/png/pokeball-64.png"
                type="image/png"
              />
              <img
                alt="Pokeball"
                class="emotion-4"
                data-testid="image-regular"
                height="64"
                loading="lazy"
                src="/assets/img/png/pokeball-64.png"
                width="64"
              />
            </picture>
            <picture>
              <source
                data-testid="image-webp"
                srcset="/assets/img/webp/pikachu-64.webp"
                type="image/webp"
              />
              <source
                data-testid="image-png"
                srcset="/assets/img/png/pikachu-64.png"
                type="image/png"
              />
              <img
                alt="Pokemon"
                class="emotion-5"
                data-testid="image-regular"
                height="56"
                loading="lazy"
                src="/assets/img/png/pikachu-64.png"
                width="56"
              />
            </picture>
          </div>
          <h2>
            Catching Pokemon . . .
          </h2>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
